tests/s3/copy_object/test_copy_object_tag.py

```diff
@@ -0,0 +1,1104 @@
+                               self.object_name2, put_etag=put_etag, copy_etag=copy_etag)
+        self.log.info("Step : Verification of etag and metadata for data integrity check")
+        self.log.info("Step 5: Copy-object with destination tag set same as source object"
+                      "in the different bucket")
+        status, response = self.s3_test_obj.copy_object(self.bucket_name1,
+                                                        self.object_name1,
+                                                        self.bucket_name2,
+                                                        self.object_name2,
+                                                        TaggingDirective='REPLACE',
+                                                        Tagging=tag_str[:-1])
+        assert_utils.assert_true(status, response)
+        copy_etag = response['CopyObjectResult']['ETag']
+        self.log.info("Step 6: Retrieving tag of a destination object %s", self.object_name2)
+        copy_resp = self.tag_obj.get_object_tags(self.bucket_name2, self.object_name2)
+        self.log.info("Retrieved tag of an object")
+        self.log.info("Step 7: Compare tagset of source and destination object")
+        assert_utils.assert_equal(len(copy_resp[1]), len(put_resp[1]), len(put_resp[1]))
+        for num in range(len(copy_resp[1])):
+            assert_utils.assert_not_equal(copy_resp[1][num]["Key"],
+                                          put_resp[1][num]["Key"], put_resp[1][num]["Key"])
+            assert_utils.assert_not_equal(copy_resp[1][num]["Value"],
+                                          put_resp[1][num]["Value"], put_resp[1][num]["Value"])
+        self.log.info("Step 7: Compared and verified tag of source and destination object"
+                      "are different and do not match")
+        self.log.info("Step :Verify tagset of destination object should match with"
+                      " tagset assigned in copy_object")
+        for num in range(len(copy_resp[1])):
+            assert_utils.assert_equal(copy_resp[1][num]["Key"],
+                                      "{}{}".format(self.key_dest, str(num)),
+                                      "{}{}".format(self.key_dest, str(num)))
+            assert_utils.assert_equal(copy_resp[1][num]["Value"],
+                                      "{}{}".format(self.value_dest, str(num)),
+                                      "{}{}".format(self.value_dest, str(num)))
+        self.log.info("Step 8: Verification of etag and metadata for data integrity check")
+        self.copy_obj_di_check(self.bucket_name1, self.object_name1, self.bucket_name2,
+                               self.object_name2, put_etag=put_etag, copy_etag=copy_etag)
+        self.log.info("Step 8: Verification of etag and metadata for data integrity check")
+        self.log.info("ENDED: copy-object with destination tag set same as source object"
```

Comment:
this should be same as of STARTED statement

Reply:
Updated